### DISCUSSION TITLE
Custom Integration Documentations

### DISCUSSION THREAD

**Discussion by bllmo** [duckdb/duckdb#5657] [Q&A]
Hi! 

Hello everyone, I'm trying to create an integration for a visualization and I can't find the necessary documentation to integrate this type of function within duckdb.

`AtomFunction`: This function takes a list of strings as input and I need to be able to do an operation for each element of the array passed in.

example:

```sql
SELECT atom_sketch(['a','b','c','d']) -- return binary
```

`AtomEstimateFunction `:  This function takes two binary inputs (two columns) and I need to be able to do `bytes.data()` within it.

example:

```sql
SELECT atom_estimate('binary_1', 'binary_2') -- return double
```

Here is the code I am using to define these functions:

```c++
namespace duckdb {
  static void AtomFunction(DataChunk &args, ExpressionState &state, Vector &result) {
    // for each of args[0]
   // return a binary
  }

  static void AtomEstimateFunction(DataChunk &args, ExpressionState &state, Vector &result) {
    // args[0] binary bytes.data()
    // args[1] binary bytes.data()
   // return double
  }

  CreateScalarFunctionInfo DatasketchesFunctions::GetThetaSketchFunction() {
	  return CreateScalarFunctionInfo(
	    ScalarFunction("atom_sketch", {LogicalType::LIST(LogicalType::VARCHAR)}, LogicalType::BLOB, AtomFunction));
  }

  CreateScalarFunctionInfo DatasketchesFunctions::GetThetaIntersectionEstimateFunction() {
	  return CreateScalarFunctionInfo(
	    ScalarFunction("atom_estimate", {LogicalType::BLOB,LogicalType::BLOB}, LogicalType::DOUBLE, AtomEstimateFunction));
  }  
}
```

I tried with `BinaryExecutor::Execute` and `TernaryExecutor::Execute` but I receive `no matching function for call to 'Execute'`

**Comment by Mytherin** [comment]
I would recommend looking at how the other functions are implemented as a reference.
